packages/expo-notifications/ios/EXNotifications/Badge/BadgeModule.swift

```diff
@@ -0,0 +1,25 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class BadgeModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoBadgeModule")
+
+    AsyncFunction("getBadgeCountAsync") { () -> Int in
+      return EXSharedApplication().applicationIconBadgeNumber
+    }
+
+    AsyncFunction("setBadgeCountAsync") { (badgeCount: Int, promise: Promise) in
+      UNUserNotificationCenter.current().getNotificationSettings { settings in

```

Comment:
There is a newer API that supports Swift concurrency, see [Apple docs](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/getnotificationsettings(completionhandler:)#Discussion)
If you use it, you don't need to use a Promise at all, see my suggestion below